File: patch_routes.py
file_php = 'D:/dsBilling/routes/web.php'
with open(file_php, 'r', encoding='utf-8') as f:
    content = f.read()

search = "Route::prefix('isp')->name('isp.')->group(function () {"
routes = """Route::prefix('isp')->name('isp.')->group(function () {
            // Voucher Templates
            Route::get('/voucher-templates', \\App\\Livewire\\ISP\\VoucherTemplate\\Index::class)->name('voucher-templates.index');
            Route::get('/voucher-templates/create', \\App\\Livewire\\ISP\\VoucherTemplate\\Editor::class)->name('voucher-templates.create');
            Route::get('/voucher-templates/import', \\App\\Livewire\\ISP\\VoucherTemplate\\Import::class)->name('voucher-templates.import');
            Route::get('/voucher-templates/{id}/edit', \\App\\Livewire\\ISP\\VoucherTemplate\\Editor::class)->name('voucher-templates.edit');
            Route::get('/voucher-templates/{id}/preview', \\App\\Livewire\\ISP\\VoucherTemplate\\Preview::class)->name('voucher-templates.preview');
            Route::get('/voucher-templates/{id}/versions', \\App\\Livewire\\ISP\\VoucherTemplate\\Versions::class)->name('voucher-templates.versions');
"""

content = content.replace(search, routes)
with open(file_php, 'w', encoding='utf-8') as f:
    f.write(content)
print("Voucher template routes added!")

File: patch_routes.php
<?php
$file = 'D:/dsBilling/routes/web.php';
$content = file_get_contents($file);

$search = <<<'PHP'
    // ==================== RESELLER PORTAL ====================
    Route::middleware(['role:reseller'])->prefix('reseller-portal')->name('reseller-portal.')->group(function () {
        Route::get('/dashboard', \App\Livewire\ResellerPortal\Dashboard::class)->name('dashboard');
    });
PHP;

$replace = <<<'PHP'
    // ==================== RESELLER PORTAL ====================
    Route::middleware(['role:reseller'])->prefix('reseller-portal')->name('reseller-portal.')->group(function () {
        Route::get('/dashboard', \App\Livewire\ResellerPortal\Dashboard::class)->name('dashboard');
        
        // Placeholder routes for missing modules
        $comingSoon = \App\Livewire\ResellerPortal\ComingSoon::class;
        
        Route::prefix('customers')->name('customers.')->group(function() use ($comingSoon) {
            Route::get('/', $comingSoon)->name('index');
            Route::get('/pppoe', $comingSoon)->name('pppoe');
            Route::get('/hotspot', $comingSoon)->name('hotspot');
            Route::get('/active', $comingSoon)->name('active');
            Route::get('/isolated', $comingSoon)->name('isolated');
        });
        
        Route::get('/service-profiles', $comingSoon)->name('service-profiles.index');
        Route::get('/pppoe-users', $comingSoon)->name('pppoe-users.index');
        Route::get('/hotspot-users', $comingSoon)->name('hotspot-users.index');
        Route::get('/vouchers', $comingSoon)->name('vouchers.index');
        Route::get('/invoices', $comingSoon)->name('invoices.index');
        Route::get('/payments', $comingSoon)->name('payments.index');
        Route::get('/reports', $comingSoon)->name('reports.index');
    });
PHP;

$content = str_replace($search, $replace, $content);
file_put_contents($file, $content);
echo "Added placeholder routes.\n";
?>
